src/exporter/ddprof_exporter.cc

```diff
@@ -91,20 +115,43 @@ DDRes ddprof_exporter_init(const ExporterInput *exporter_input,
   }
 
   if (exporter->_agent) {
-    exporter->_url =
-        alloc_url_agent("http://", exporter_input->host, exporter_input->port);
+    const char *port_str = exporter_input->port;
```

Comment:
This should be probably be extracted to its own function.
I would simplify the logic by sopping mixing url and port:
* either a url is given in input, in that case it's expected to be a full url (with protocol and port), in that host and port are ignored
* either no url is given in input and url is built from 'http', host and port

That way you can even stop distinguishing between agent and agentless.

Reply:
This is a good idea, though we would need to check if endpoint is reachable on startup,
Adding this permissive config might be bad for reverting it later.